firstvoices/backend/views/data_views.py

```diff
@@ -44,25 +52,65 @@ class SitesDataViewSet(
     viewsets.GenericViewSet,
 ):
     http_method_names = ["get"]
-    serializer_class = SiteDataSerializer
-    pagination_class = None
+    serializer_class = DictionaryEntryDataSerializer
+    pagination_class = DictionaryEntryPaginator
+    renderer_classes = [SnakeCaseJSONRenderer, BrowsableAPIRenderer]
 
     def get_queryset(self):
-        sites = (
+        site = self.get_validated_site()
+        return DictionaryEntry.objects.filter(site__id__in=site).prefetch_related(
+            "part_of_speech",
+            "site",
+            Prefetch(
+                "related_audio",
+                queryset=Audio.objects.all()
+                .select_related(
+                    "original",
+                )
+                .prefetch_related(
+                    "speakers",
+                ),
+            ),
+            Prefetch(
+                "related_images",
+                queryset=Image.objects.all().select_related(
+                    "original",
+                ),
+            ),
```

Comment:
Do we also need to prefetch videos?

Reply:
The API doesn't currently return video data as that was not in the ticket requirements. It could be added if needed though.